frontend/src/admin/lessons/EditLesson.tsx

```diff
@@ -1,86 +1,204 @@
-import React from 'react';
-import { Nav, Container, FormInput, Button } from '@components/';
-import { Link } from 'react-router-dom';
+import React, { useState, useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { useForm, SubmitHandler, useFieldArray } from 'react-hook-form';
+
+import { Link, useNavigate, useParams } from 'react-router-dom';
+import { Nav, Container, FormInput, Button, Loader } from '@components/';
+import { getSpecificCategory } from '@api/CategoryApi';
+import { Category, setIsError, setIsLoading } from '@store/category';
+import { RootState } from '@store/store';
+import { getLesson, updateLesson } from '@api/LessonApi';
+import { Choice } from '@store/lessons';
+
+type Inputs = {
+  id: number;
+  word: string;
+  correct_answer: [];
+  choices: [];
+};
 
 export default function EditLesson() {
   const currentPath = window.location.pathname;
+  const navigate = useNavigate();
+  const state = useSelector((state: RootState) => state);
+  const [errorMsg, setErrorMsg] = useState('');
+  const [categoryItem, setCategoryItem] = useState({
+    id: 0,
+    name: '',
+    description: '',
+  });
+  const {
+    register,
+    reset,
+    handleSubmit,
+    formState: { errors },
+  } = useForm();
+  const { category_id, lesson_id } = useParams();
+  const dispatch = useDispatch();
+
+  const _getSpecificCategory = (category_id: number) => {
+    getSpecificCategory(category_id)
+      .then((response) => {
+        setCategoryItem(response.data.data);
+      })
+      .catch(() => {
+        dispatch(setIsError(true));
+      });
+  };
+
+  const _getSpecificLesson = (lesson_id: number) => {
+    getLesson(lesson_id)
+      .then((response) => {
+        const choices = response.data.data.choices;
+        reset({
+          id: response.data.data.id,
+          word: response.data.data.word,
+          correct_answer: choices.filter(
+            (choice: { name: string; is_correct: boolean }) => choice.is_correct
+          ),
```

Comment:
You can do de-structuring here to avoid some redundant code

```suggestion
        const { choices, id, word } = response.data.data;
        reset({
          id: id,
          word: word,
          correct_answer: choices.filter(
            (choice: { name: string; is_correct: boolean }) => choice.is_correct
          ),
```